torch/fx/graph.py

```diff
@@ -502,13 +537,15 @@ def type_repr(o : Any):
                 body.append(
                     f'{node.name} = {_format_target(repr(node.args[0]), node.target)}'
                     f'({_format_args(node.args[1:], node.kwargs)})\n')
+                delete_unused_values(node)
```

Comment:
the control structure here is confusing. I'd expect delete_unused_values to only get issued once, but I see it inside nested if statements. Maybe refactor `for node in self.nodes: emitNode(node); emit_delete_unused_values(node);`